include/swift/Basic/SourceLoc.h

```diff
@@ -130,7 +130,9 @@ class CharSourceRange {
 
 public:
   /// \brief Constructs an invalid range.
-  CharSourceRange() {}
+  CharSourceRange() = default;
+
+  CharSourceRange &operator=(const CharSourceRange &) = default;
```

Comment:
[Rule of Zero](https://en.cppreference.com/w/cpp/language/rule_of_three) says that we should not declare this explicitly. If you think it's important, we should follow the Rule of Five instead.

Reply:
`CharSourceRange() {}` is effectively `CharSourceRange() = default`, right? If so, then yes, I can flush out the rule-of-five.